interface/name_decorator/base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'name_decorator'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
```

Comment:
- [ ] it would be better if you make the nameable readable and writeable by adding `attr_accessor :nameable` to the class